### PR TITLE
Fix IO::Memory#gets_to_end to consume the IO

### PR DESCRIPTION
**Pull request by jhass**

### CODE REVIEW

File: src/io/memory.cr [hunk 1]
```diff
@@ -225,7 +225,9 @@ class IO::Memory
     if pos == @bytesize
       ""
     else
-      String.new(@buffer + @pos, @bytesize - @pos)
+      String.new(@buffer + @pos, @bytesize - @pos).tap {
```
Comment (RX14) [inline src/io/memory.cr]: Isn't the style rule that multi-line blocks are always `do`/`end`?

Comment (jhass) [inline src/io/memory.cr]: Uh well my personal style certainly isn't, the formatter doesn't seem to enforce it and https://crystal-lang.org/docs/conventions/coding_style.html doesn't mention anything about it.

Comment (RX14) [inline src/io/memory.cr]: The ruby style guide discourages it for the good reason that it encourages chaining methods on the return values of multi-line blocks. And I tend to agree with it. I'm not sure what the core team thinks though.

Comment (jhass) [inline src/io/memory.cr]: I roughly follow [Weirich's rule](https://web.archive.org/web/20140221124509/http://onestepback.org/index.cgi/Tech/Ruby/BraceVsDoEnd.rdoc) personally.

Comment (RX14) [inline src/io/memory.cr]: In my opinion, the stdlib should stick to one rule, and that rule should be easy to enforce.

**Comment by asterite** [comment]
@jhass Thanks!

**Pull request merged**